modules/s3-articfact/main.tf

```
@@ -0,0 +1,380 @@
+# Object versioning is enabled with a 30 day noncurrent version expiration policy and transitions
+# are enabled with a 30 day transition to STANDARD_IA and a 90 day transition to GLACIER.
+# Partial lifecycle management enables only object versioning with a 30 day noncurrent version
+# expiration policy. The full lifecycle management variable is a boolean and is defaulted to true.
+#
+# The bucket replication configuration creates a bucket in a different region. The replica
+# bucket will be encrypted with the same "CMK" as the primary bucket. The replica bucket defines
+# a bucket policy that allows the primary bucket to replicate objects to it.
+#
+# The module includes the following:
+#
+# - Primary S3 Bucket
+# - Primary S3 Bucket ACL
+# - Primary S3 Bucket Versioning
+# - Primary S3 Bucket Lifecycle Management (Full or Partial)
+# - Primary S3 Bucket Public Access Configuration
+# - Primary S3 Bucket Server Side Encryption Configuration
+# - Replica S3 Bucket
+# - Replica S3 Bucket ACL
+# - Replica S3 Bucket Versioning
+# - Replica S3 Bucket Public Access Configuration
+# - Replica S3 Bucket Server Side Encryption Configuration
+# - Primary IAM Policy Document
+# - Primary IAM Policy
+# - Replica IAM Policy Document
+# - Replica IAM Role
+# - Replica IAM Role Policy Attachment
+#
+# -------------------------------------------------------------------------------------
+
+
+# -------------------------------------------
+# SET TERRAFORM REQUIREMENTS TO RUN MODULE
+# -------------------------------------------
+
+terraform {
+  required_version = ">= 1.5.5"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 5.0"
+    }
+  }
+}
+
+# -------------------------------------------
+# SET PROVIDER ALIAS FOR REPLICATION
+# -------------------------------------------
+
+provider "aws" {
+  region = var.priamry_region
+}
+
+provider "aws" {
+  alias  = "replica"
+  region = var.replica_region
+}
+
+
+# ------------------------------------------------------------
+
+# THE FOLLOWING SECTION IS USED TO CREATE THE PRIMARY S3 BUCKET
+
+# AND ASSOCIATED RESOURCES (LIFECYCLE MANAGEMENT, VERSIONING).
+
+# ------------------------------------------------------------
+
+
+# -------------------------------------------
+# CREATE PRIMARY S3 BUCKET
+# -------------------------------------------
+
+resource "aws_s3_bucket" "primary" {
+  bucket = var.primary_bucket_name
+}
+
+# -------------------------------------------
+# CREATE PRIMARY S3 BUCKET POLICY
+# -------------------------------------------
+
+resource "aws_s3_bucket_acl" "primary" {
+  bucket = aws_s3_bucket.primary.id
+  acl    = var.pimary_bucket_acl
+}
+
+# -------------------------------------------
+# ENABLE PRIMARY OBJECT VERSIONING
+# -------------------------------------------
+
+resource "aws_s3_bucket_versioning" "primary" {
+  bucket = aws_s3_bucket.primary.id
+  versioning_configuration {
+    status = var.bucket_versioning_configuration_status
```

This goes with the change mentioned above.

```suggestion
    status = "ENABLED"
```